File: csharp/Leet/Today3/NonOverlappingIntervals.cs
// https://leetcode.com/problems/non-overlapping-intervals/description/
namespace Today3.NonOverlappingIntervals;

public class Solution
{
    public int EraseOverlapIntervals(int[][] intervals)
    {
        Array.Sort(intervals, (a, b) => a[1].CompareTo(b[1]));
        var prev = intervals[0];
        var count = 1;
        for (int i = 1; i < intervals.Length; i++)
        {
            if (prev[1] <= intervals[i][0])
            {
                count++;
                prev = intervals[i];
            }
        }
        return intervals.Length - count;
    }
}

File: python/today41/non_overlapping_intervals.py
# https://leetcode.com/problems/non-overlapping-intervals/description/
from typing import List


class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[1])

        count = 1
        prev = 0
        for i in range(1, len(intervals)):
            if intervals[prev][1] <= intervals[i][0]:
                count += 1
                prev = i

        return len(intervals) - count
